src/snmpd/patch-5.7.3+dfsg/0010-boot_reason.patch

```diff
@@ -0,0 +1,60 @@
+--- net-snmp-5.7.3+dfsg/agent/snmpd.c.org	2020-01-05 23:11:50.926268000 -0800
++++ net-snmp-5.7.3+dfsg/agent/snmpd.c	2020-01-05 23:11:56.092916000 -0800
+@@ -31,7 +31,7 @@
+  * *****************************************************************
+  */
+ /*
+- * Copyright � 2003 Sun Microsystems, Inc. All rights reserved.
++ * Copyright ? 2003 Sun Microsystems, Inc. All rights reserved.
+  * Use is subject to license terms specified in the COPYING file
+  * distributed with the Net-SNMP package.
+  */
+@@ -413,6 +413,30 @@
+      */
+ }
+ 
++int get_reboot_reason(void)
++{
++    FILE *fp;
++    char *ch;
++    char buf[1000];
++    int i=0;
++    char *reboot_reason[] = {"User issued 'warm-reboot'",
++                              NULL};
++    fp = fopen("/host/reboot-cause/previous-reboot-cause.txt", "r");
```

Comment:
>fopen [](start = 10, length = 5)

This is feature relating to sonic. Is it possible to implement outside snmpd, like in sonic-snmpagent?